### PR TITLE
feat: update docs

### PR DESCRIPTION
**Pull request by Milton-Ch**
- Updated documentation according to observations.
- Added notes to all flows for better understanding

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+
+```
+{
+  "allow_spontaneous_scopes": false,
+  "application_type": "web",
+  "rpt_as_jwt": false,
+  "registration_client_uri": "https://jenkins-dev1.jans.io/jans-auth/restv1/register?client_id=27975f1c-eee6-4bf8-b393-5fb47d44c566",
+  "tls_client_auth_subject_dn": "",
+  "run_introspection_script_before_jwt_creation": false,
+  "registration_access_token": "ca1e8640-d59a-4603-a42c-936e69385101",
```
Comment (nynymike) [inline README.md]: You can't check an access token value into github, you need to remove this

File: README.md [hunk 2]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+```
+
+**Response**
+
+```
+{
+  "allow_spontaneous_scopes": false,
+  "application_type": "web",
+  "rpt_as_jwt": false,
+  "registration_client_uri": "https://jenkins-dev1.jans.io/jans-auth/restv1/register?client_id=27975f1c-eee6-4bf8-b393-5fb47d44c566",
```
Comment (nynymike) [inline README.md]: please do not checkin specific hostnames like this... it's an actual server isn't it?

File: README.md [hunk 3]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+Run these instructions on the server where you have `Janssen` or `Gluu` installed:
 
-Follow the steps below:
-
-- Copy (SCP/SFTP) the gama file of this project to a location in your `Jans Server`
-- Connect (SSH) to your `Jans Server` and open TUI: `python3 /opt/jans/jans-cli/jans_cli_tui.py`
-- Navigate to the `Agama` tab and then select `"Upload project"`. Choose the gama file
+- Download the
+  latest [agama-passkey.gama](https://github.com/GluuFederation/agama-passkey/releases/latest/download/agama-passkey.gama), you can use `wget`
+```
```
Comment (nynymike) [inline README.md]: use ```shell so copy/paste icon is added

File: README.md [hunk 4]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+wget https://github.com/GluuFederation/agama-passkey/releases/latest/download/agama-passkey.gama
+```
+- Open TUI: `python3 /opt/jans/jans-cli/jans_cli_tui.py`
```
Comment (nynymike) [inline README.md]: These are jans instructions with nothing specific to your project. Remove this whole section, or just reference the Jans docs about agama project deployment. If you have any ideas about how to make the official docs better for Agama project deployment, make a PR for Jans docs.

File: README.md [hunk 5]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+  "scope": [
+    "https://jans.io/scim/fido2.read",
+    "https://jans.io/scim/fido2.write"
+  ],
+  "grant_types": [
+    "client_credentials"
+  ],
+  "token_endpoint_auth_method": "client_secret_basic",
+  "response_types": [
+    "code"
```
Comment (nynymike) [inline README.md]: why does a backchannel client need response_type=code? It doesn't... remove response_type.

File: README.md [hunk 6]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+  "frontchannel_logout_session_required": false,
+  "client_secret_expires_at": 1710555708,
+  "access_token_signing_alg": "RS256",
+  "response_types": [
+    "code"
+  ]
+}
+```
+
+or you can use the `TUI`
```
Comment (nynymike) [inline README.md]: reference jans docs... don't repeat jans docs ...

File: README.md [hunk 7]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
 
 ### Deployment
```
Comment (nynymike) [inline README.md:39]: I really don't even know why you have a deployment section at all... this should be in Jans docs.

File: README.md [hunk 8]
```diff
@@ -23,7 +23,9 @@ To deploy this project we need to meet the requirements.
 
 ### Requirements
 
-1. Running instance of `Jans Auth Server`, `Jans Fido2`, `Jans Casa` and `Jans Scim`
+1. Running instance of
+   - `Jans Auth Server`-
+   - `Jans Fido2`, `Jans Casa` and `Jans Scim`
```
Comment (nynymike) [inline README.md]: Why is Auth Server on one line, but FIDO, Casa and SCIM are on one line?

File: README.md [hunk 9]
```diff
@@ -57,28 +59,106 @@ systemctl restart jans-auth
...
+
+You can create the client using the registration web service.
+
+**Request**
+
+```
+curl --location 'https://YOUR_DOMAIN/jans-auth/restv1/register' \
+--header 'Content-Type: application/json' \
+--data '{
+  "client_name": "Scim Agama Client",
```
Comment (nynymike) [inline README.md]: SCIM is all caps... it's an acronym

**Review by nynymike — CHANGES_REQUESTED** [review]
* I'd like to see a section that lists the flows, like [here](https://github.com/GluuFederation/agama-OATH-TOTP?tab=readme-ov-file#flows) 

* There is no JSON config for this project? If there is, please document like [here](/opt/jans/jetty/jans-auth/webapps/jans-auth.xml) 

* I'd like to see the "Add Java dependencies" reduced to 4 lines:
1. cd /opt/jans/jetty/jans-auth/custom/libs
2. wget https://github.com/GluuFederation/agama-passkey/releases/latest/download/agama-passkey-custom.jar
3. If you are not using *.jar, update extractClasspath in /opt/jans/jetty/jans-auth/webapps/jans-auth.xml
4. Restart server

* Net-net, I would remove all the deployment stuff except adding the SCIM client. But you can read my comments about all the stuff that was problematic in there.  And if somehow the Jans docs aren't working for agama project deployment, please make a PR to fix the docs so all the projects benefit. If you were going to put the Agama Project Deployment in there, I'd really like to see you use one CLI command then all that TUI stuff.

File: README.md (hunk 4 above)
Comment (Milton-Ch) [reply README.md]: You are right, so I just remove all those instructions and leave the .gif?
Or do I totally remove everything and place a reference link to a deployment of an agama project, like this https://docs.jans.io/v1.1.0/admin/developer/agama/quick-start-using-agama-lab/#deploy-agama-project?

File: README.md (hunk 1 above)
Comment (Milton-Ch) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (Milton-Ch) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (Milton-Ch) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (Milton-Ch) [reply README.md]: Done

File: README.md (hunk 6 above)
Comment (Milton-Ch) [reply README.md]: Remove this section, the first example of client creation should suffice.

File: README.md (hunk 8 above)
Comment (Milton-Ch) [reply README.md]: Solved

File: README.md (hunk 9 above)
Comment (Milton-Ch) [reply README.md]: Solved

**Review by nynymike — APPROVED** [review]
ok

**Review by yuriyz — APPROVED** [review]

**Pull request merged**